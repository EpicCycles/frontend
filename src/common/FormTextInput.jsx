/* eslint-disable jsx-a11y/label-has-for */
import React from 'react';
import { Icon } from 'semantic-ui-react';
import * as PropTypes from 'prop-types';

const FormTextInput = props => {
  const {
    value,
    className,
    dataType,
    disabled,
    error,
    fieldName,
    inputClass,
    label,
    list,
    maxLength,
    onChange,
    onClick,
    onKeyPress,
    placeholder,
    title,
    size,
  } = props;
  return (
    <div className={`row ${className}${error ? ' error' : ''}`}>
      {label && <label>{label}</label>}
      <input
        className={`${inputClass} ${error ? ' error' : ''}`}
        type={dataType}
        autoComplete="off"
        placeholder={placeholder}
        title={`${title} ${error}`}
        name={fieldName}
        key={fieldName}
        onChange={event => onChange(fieldName, event.target.value)}
        value={value}
        size={size}
        list={list}
        disabled={disabled}
        maxLength={maxLength || size}
        onKeyPress={onKeyPress}
      />
      {value && onClick && (
        <span className="clearInput">
          <Icon
            name="remove"
            key={`remove${fieldName}`}
            id={`remove${fieldName}`}
            size="small"
            circular
            link
            onClick={() => onClick(fieldName)}
          />
        </span>
      )}
    </div>
  );
};

FormTextInput.defaultProps = {
  className: '',
  error: '',
  inputClass: '',
  label: '',
  dataType: 'text',
  placeholder: '',
  title: '',
  value: '',
  list: '',
  size: 30,
};
FormTextInput.propTypes = {
  className: PropTypes.string,
  error: PropTypes.string,
  label: PropTypes.string,
  dataType: PropTypes.string,
  placeholder: PropTypes.string,
  title: PropTypes.string,
  value: PropTypes.string,
  list: PropTypes.string,
  fieldName: PropTypes.string.isRequired,
  inputClass: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  onClick: PropTypes.func,
  onKeyPress: PropTypes.func,
  size: PropTypes.number,
  maxLength: PropTypes.number,
  disabled: PropTypes.bool,
};
export default FormTextInput;
