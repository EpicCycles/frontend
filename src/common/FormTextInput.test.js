import FormTextInput from './FormTextInput';

describe('FormTextInput', () => {
  it('renders the form text input correctly', () => {
    const input = shallow(
      <FormTextInput
        id="1234"
        label="Test"
        fieldName="testfield"
        placeholder="Please enter your loyalty number"
        onChange={jest.fn()}
      />,
    );
    expect(toJson(input)).toMatchSnapshot();
  });
  it('renders the form numeric input correctly', () => {
    const input = shallow(
      <FormTextInput
        id="1234"
        label="Test"
        fieldName="testfield"
        placeholder="Please enter your loyalty number"
        onChange={jest.fn()}
        dataType="number"
      />,
    );
    expect(toJson(input)).toMatchSnapshot();
  });

  it('renders an error when present', () => {
    const input = shallow(
      <FormTextInput
        id="5678"
        label="Test"
        fieldName="testfield"
        placeholder="Please enter your loyalty number"
        onChange={jest.fn()}
        error="Incorrect loyalty number"
      />,
    );
    expect(toJson(input)).toMatchSnapshot();
  });

  it('should call onClearEmail when the button is clicked', () => {
    const onChange = jest.fn();
    const value = 'test@johnlewis.co.uk';
    const fieldName = 'email';
    const onClearEmail = jest.fn();
    const input = shallow(
      <FormTextInput
        onClick={onClearEmail}
        fieldName="testfield"
        onChange={onChange}
        value={value}
        fieldName={fieldName}
      />,
    );
    expect(toJson(input)).toMatchSnapshot();
    expect(input.find('#removeemail')).toHaveLength(1);
    input.find('#removeemail').simulate('click');
    expect(onClearEmail.mock.calls).toHaveLength(1);
  });

  it('should call onChange when an input is entered', () => {
    const onChange = jest.fn();
    const value = 'test@johnlewis.co.uk';
    const fieldName = 'email';
    const onClearEmail = jest.fn();
    const input = shallow(
      <FormTextInput
        onClick={onClearEmail}
        fieldName="testfield"
        onChange={onChange}
        value={value}
        fieldName={fieldName}
      />,
    );
    expect(toJson(input)).toMatchSnapshot();
    input.find('input').simulate('change', { target: { value: 1234567890123456 } });
    expect(onChange.mock.calls).toHaveLength(1);
  });
});
