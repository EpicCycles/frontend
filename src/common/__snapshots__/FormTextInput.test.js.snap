// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormTextInput renders an error when present 1`] = `
<div
  className="row  error"
>
  <label>
    Test
  </label>
  <input
    autoComplete="off"
    className="  error"
    list=""
    maxLength={30}
    onChange={[Function]}
    placeholder="Please enter your loyalty number"
    size={30}
    title=" Incorrect loyalty number"
    type="text"
    value=""
  />
</div>
`;

exports[`FormTextInput renders the form numeric input correctly 1`] = `
<div
  className="row "
>
  <label>
    Test
  </label>
  <input
    autoComplete="off"
    className=" "
    list=""
    maxLength={30}
    onChange={[Function]}
    placeholder="Please enter your loyalty number"
    size={30}
    title=" "
    type="number"
    value=""
  />
</div>
`;

exports[`FormTextInput renders the form text input correctly 1`] = `
<div
  className="row "
>
  <label>
    Test
  </label>
  <input
    autoComplete="off"
    className=" "
    list=""
    maxLength={30}
    onChange={[Function]}
    placeholder="Please enter your loyalty number"
    size={30}
    title=" "
    type="text"
    value=""
  />
</div>
`;

exports[`FormTextInput should call onChange when an input is entered 1`] = `
<div
  className="row "
>
  <input
    autoComplete="off"
    className=" "
    key="email"
    list=""
    maxLength={30}
    name="email"
    onChange={[Function]}
    placeholder=""
    size={30}
    title=" "
    type="text"
    value="test@johnlewis.co.uk"
  />
  <span
    className="clearInput"
  >
    <Icon
      as="i"
      circular={true}
      key="removeemail"
      link={true}
      name="remove"
      onClick={[Function]}
      size="small"
    />
  </span>
</div>
`;
